### ISSUE TITLE
When You Need To Reload - page reload requirement affordance

### ISSUE THREAD

**Issue by lansingthomas** [atlas-engineer/nyxt#3178]
**Purpose of request:**
sometimes a page reload is the right thing to do, and the user needs to know about it.

**Describe your proposed change:**
1. For cases where a page reload is required to propagate changes, (certain modes, certain settings..) I propose adding a color change to the hover area of the existing `reload` button in the Status Buffer.
2. for the `toggle-modes` menu, add a helpful echo to clarify what is happening as different modes are marked/unmarked

**Why do you believe this to be an improvement?**
1. Small ways to show state changes for our system help orient users
2. the change will not get in the way of other UI elements, it is just a state change for an existing element.
3. clarifies the sometimes confusing system states in the `toggle-modes` menu.
4. provides some error prevention for changes made in the Common Settings page. 

**Additional context/User story:**
I have not thought through every place this might apply, and I suspect there will be times where prompting the user for a page reload is helpful.

**Screenshots/Mock ups:**

![prompt buffer (open) component](https://github.com/atlas-engineer/nyxt/assets/6666079/1d24a301-d5f0-4e51-827f-19fdbf524eae)

---

### Pre-Build Checklist:
- [x] Discussion has occurred.
- [x] One or more developers have signed off on this change.
- [x] One or more user researchers have signed off on this change.

**Comment by lansingthomas** [comment]
@aartaka this is related to our upcoming conversation about checkboxes :D

**Comment by lansingthomas** [comment]
The color for this `state: reload-required` should be _action+_ from our palette ^^
- `#72CDFE`
- in dark theme this is `#4C21AB`

**Comment by lansingthomas** [comment]
Hi @aartaka,
do you need any more information for this to happen?

I think this a resolution for this issue will be helpful for the upcoming (soon) Common Settings page.

**Comment by aartaka** [comment]
No, it's clear enough. But I'll de-assign: I'm getting back to more pressing things. @jmercouris, @hgluka, any taker?

**Comment by lansingthomas** [comment]
> the Common Settings page requires a restart, not just a reload. 

Ah yes. 

> a note/popup/`if-confirm` prompt 

This makes sense... let's keep the Restart prompt separate for now. I'll open an issue to capture the idea.

**Comment by lansingthomas** [comment]
Fair enough. I am solution agnostic for this problem: **Poor visibility of system state when a page requires reload**, (and when the app requires restart)

I am seeing [3] potential remedies, which could happen individually, in combination, or not at all.

1. light up the reload button when appropriate
2. echo in the message buffer when appropriate
3. reduce or remove the cases where a page reload or an app restart are required

@aadcg & @jmercouris (if alive) Please help by deciding a course of action for this one. 

Or tell me to ignore it -- just please acknowledge that it is annoying to wonder "is this broken? or do i need to reload...?"

I am not able to estimate the difficulty of implementation the way you two can. My perception is that Nyxt will require re-loads and re-starts for a long time... dunno. 

Thank you!

P.S.
If working backwards is what it takes to make nyxt more usable then let's go backwards _sdrawkcab og s'tel_ :)

**Comment by aadcg** [comment]
Similar to #3193.
